modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -657,7 +657,11 @@ def lazy_map_partitions(
         NumPy array
             An array of partitions
         """
-        preprocessed_map_func = cls.preprocess_func(map_func)
+        preprocessed_map_func = (
+            map_func
+            if cls._execution_wrapper.is_future(map_func)
```

Comment:
When do we pass map_func as a future in this method?

Reply:
Here https://github.com/modin-project/modin/pull/6972/files#diff-718e1f98ecbed40356292a477fc27bafb81e62c796491c81e398d5b397d6d69eR4668 the remote function is put to the wrapper. After tat, an object reference is passed to this function.

Reply:
We should update docstring.